backend/flask_api.py

```diff
@@ -0,0 +1,131 @@
+import os
+import sys
+from werkzeug.utils import redirect
+from flask.helpers import flash
+from flask_cors import CORS
+from flask import Flask, request, jsonify, make_response, redirect, url_for, render_template
+from werkzeug.utils import secure_filename
+from pymongo import MongoClient
+from manifest_validator import ManifestValidator
+import time, json5, uuid, json
+import elastic
+from elasticsearch import Elasticsearch
+import uploader
+
+
+client=MongoClient('mongodb:27017')
+db=client.knexDB
+coll=db.projects
+
+schema = open("manifest_schema.json")
+validator = ManifestValidator(schema)
+
+app=Flask(__name__)
+CORS(app)
+
+ALLOWED_EXTENSIONS = set(['txt', 'json', 'json5'])
+app.config['UPLOAD_FOLDER'] = ''
+app.config['MAX_CONTENT_PATH'] = 1000000; #100.000 byte = 100kb 
+
+
+@app.route('/', methods=['GET'])
+def index():
+    return make_response('', 404)
+
+# receive manifest as a jsonstring
+# returns new id
+@app.route('/api/projects', methods=['POST'])
+def add_project():
+    manifest=request.json
```

Comment:
This was the code path you were actually suppose to test and fix and this looks very similar to `save_file_to_db`.
Could you merge the two? We actually need to file upload, the frontend will already read the file into a json string.